libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/service/MapboxTripService.kt

```diff
@@ -62,13 +63,14 @@ internal class MapboxTripService(
         tripNotification,
         {
             try {
-                applicationContext.startService(intent)
-            } catch (e: IllegalStateException) {
                 if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                     applicationContext.startForegroundService(intent)
                 } else {
-                    throw e
+                    applicationContext.startService(intent)
                 }
+            } catch (e: IllegalStateException) {
+                logW("Could not start foreground service with exception: $e. Note that you " +
+                    "should invoke `MapboxNavigation#startTripSession` from foreground only.")
```

Comment:
This change can put the SDK in somewhat of an undefined state because app requested to start with a foreground service that wasn't started, but the rest of the SDK operates as if it did.

I'd propose removing the catch clause.

Reply:
OK, imakes sense.

Reply:
Or we can improve that. Because now if we don't start the service, we still set the flag to true. It's kind of inconsistent. How about I merge only the docs here and open a separate PR with a proposed solution?

Reply:
Sounds good :+1: